Spruce- Rebecca

rzuick opened the pull request.

rzuick committed:
created and rendered instances of Square for board

rzuick committed:
made changes to onClickCallback function in app.js

rzuick committed:
created function to check for winner

rzuick committed:
created checkforwinner method

rzuick committed:
modified check for winner method

rzuick committed:
updated check for winner function to display ties

rzuick committed:
updated square clicking so a square cannot be clicked again; passing all tests

rzuick committed:
added comment

rzuick committed:
made some styling changes

rzuick committed:
attempting to deploy app

audreyandoy commented on .eslintrc.json:
```diff
@@ -12,7 +12,6 @@
     "plugin:react/recommended",
     "plugin:jsx-a11y/recommended",
     "plugin:react-hooks/recommended",
-    "plugin:jest/recommended",
```
Hm.... why did you remove jest?

audreyandoy commented on src/App.css:
```diff
@@ -3,14 +3,19 @@
 .App-header {
-  background-color: #282c34;
+  background-color: #230535;
   min-height: 10vh;
   display: flex;
   flex-direction: column;
   align-items: center;
   justify-content: center;
   font-size: calc(10px + 2vmin);
-  color: white;
+  font-family: cursive;
+  color: rgb(240, 233, 102);
   margin-bottom: 2em;
```
Love the CSS additions to this project! The purple and gold are giving me regal vibes 👑

audreyandoy commented on src/components/Board.js:
```diff
@@ -3,29 +3,38 @@ import './Board.css';
 
 const Board = ({ squares, onClickCallback }) => {
   const squareList = generateSquareComponents(squares, onClickCallback);
   console.log(squareList);
```
When pushing code to be reviewed, we should eliminate any print statements or console logs
```suggestion

```

audreyandoy commented on src/components/Square.js:
```diff
@@ -1,19 +1,20 @@
-  >
-    {props.value}
-  </button>
-}
+  return (
+    <button
+      className="square"
+      onClick={() => {
+        props.onClickCallback(props.id);
+      }}
+    >
+      {props.value}
+    </button>
+  );
+};
```
Great use of an anonymous function!

We could use object destructing so we can reference props by their key name instead.

```suggestion
const Square = ( {value, id, onClickCallback, key} ) => {
    return (
    <button
      className="square"
      onClick={() => {
        onClickCallback(id);
      }}
    >
      {value}
    </button>
  );
};

```

audreyandoy commented on src/App.js:
```diff
@@ -21,45 +21,116 @@ const generateSquares = () => {
+      } else if (player === PLAYER_2) {
+        changePlayer(PLAYER_1);
+      }
+    }
```
This is a great spot to use a ternary operator

```suggestion
    if (!winner) {
     changePlayer( if player === PLAYER_1 ? PLAYER_2 : PLAYER_1
    }
```

audreyandoy commented on src/App.js:
```diff
@@ -21,45 +21,116 @@ const generateSquares = () => {
+    checkWinner('');
+    winnerText = `Winner is ${winner}`;
+    setSquares(newBoard);
+    changePlayer(PLAYER_1);
```
We can pass in the function call into `setSquares` directly
```suggestion
    checkWinner('');
    winnerText = `Winner is ${winner}`;
    setSquares(generateSquares());
    changePlayer(PLAYER_1);
```

audreyandoy commented on src/App.js:
```diff
@@ -21,45 +21,116 @@ const generateSquares = () => {
   };
 
+  const onClickCallback = (id) => {
+    let square = squares.map((oneSquareArray) => {
+      for (const insideSquare of oneSquareArray) {
+        if (insideSquare.id === id) {
+          if (insideSquare.value) {
+            continue;
+          } else {
+            insideSquare.value = player;
+          }
+        }
+      }
+      return oneSquareArray;
+    });
```
Great use of `map` to iterate and return a new array!

We can combine the conditionals in the inner loop using compound expressions and the not operator `!` like so:

```suggestion
    let square = squares.map((oneSquareArray) => {
      for (const insideSquare of oneSquareArray) {
        if (insideSquare.id === id && !insideSqaure.value) {
            insideSquare.value = player;
        }
      }
      return oneSquareArray;
    });
```

audreyandoy reviewed (commented):
Great work Rebecca! You met all the learnings goals for this project,  passed all the tests, and experimented with CSS! Love the purple and yellow/gold theme 💜⭐️

I added feedback to your PR on ways to refactor your project.

Project Grade: 🟢

rzuick committed:
Update src/App.js

Refactoring square variable to eliminate unnecessary else statement and use of "continue"

Co-authored-by: Audrey Andoy  <audreylandoy@gmail.com>

rzuick committed:
Update src/App.js

removed unnecessary variable declaration and assignment to call generateSquares function directly in state

Co-authored-by: Audrey Andoy  <audreylandoy@gmail.com>

rzuick committed:
Update src/App.js

suggestion to use ternary here was accepted. it is logical and easier to read than the extra else if statement

Co-authored-by: Audrey Andoy  <audreylandoy@gmail.com>
